src/org/labkey/targetedms/MsDataSourceUtil.java

```diff
@@ -0,0 +1,1243 @@
+package org.labkey.targetedms;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.labkey.api.data.ColumnInfo;
+import org.labkey.api.data.CompareType;
+import org.labkey.api.data.Container;
+import org.labkey.api.data.ContainerManager;
+import org.labkey.api.data.SQLFragment;
+import org.labkey.api.data.SimpleFilter;
+import org.labkey.api.data.Table;
+import org.labkey.api.data.TableInfo;
+import org.labkey.api.data.TableSelector;
+import org.labkey.api.data.dialect.SqlDialect;
+import org.labkey.api.exp.ExperimentException;
+import org.labkey.api.exp.Lsid;
+import org.labkey.api.exp.api.DataType;
+import org.labkey.api.exp.api.ExpData;
+import org.labkey.api.exp.api.ExperimentService;
+import org.labkey.api.files.FileContentService;
+import org.labkey.api.module.ModuleLoader;
+import org.labkey.api.query.FieldKey;
+import org.labkey.api.security.User;
+import org.labkey.api.targetedms.ISampleFile;
+import org.labkey.api.targetedms.TargetedMSService;
+import org.labkey.api.util.FileUtil;
+import org.labkey.api.util.JunitUtil;
+import org.labkey.api.util.TestContext;
+import org.labkey.targetedms.parser.Instrument;
+import org.labkey.targetedms.parser.PsiInstruments;
+import org.labkey.targetedms.parser.SampleFile;
+import org.labkey.targetedms.query.InstrumentManager;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+
+public class MsDataSourceUtil
```

Comment:
This a big class. I recommend creating a new package under org.labkey.targetedms (.raw, maybe), moving this into it, and pulling all of the nested classes into their own top-level classes.

Reply:
Done. Moved to org.labkey.targetedms.datasource